### ISSUE TITLE
FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory

### ISSUE THREAD

**Issue by HiImVINK** [codetheweb/muse#884]
**What's happening?**
I have written a kubernetes deployment using the docker image. Everything seems to be working except for the following.

When i try to play a playlist contain quite a lot of songs (Youtube music). The pod crashes (Out of Memory). 
The pod is allocated to use more than 8GB of memory.

What is causing this? Should i increase the cache size?

**Logs**

```
yarn run v1.22.19
$ npm run env:set-database-url -- tsx src/scripts/migrate-and-start.ts
> muse@2.1.7 env:set-database-url
> tsx src/scripts/run-with-database-url.ts
========================================
========================================
================= muse =================
========================================
=========== Made with 🎶 by ============
==== https://github.com/codetheweb =====
========================================
========================================
Running version 2.1.7 built on 9/20/2022 (commit abdae1a6449d0e3911b45c0dfde3f4b184c17ae0)
🔧 Need help? https://github.com/codetheweb/muse/discussions/new
🐛 Bug? https://github.com/codetheweb/muse/issues/new
✨ New feature? https://github.com/codetheweb/muse/issues/new
☕ Discussion? https://github.com/codetheweb/muse/discussions/new
💰 Help me? https://www.paypal.me/codetheweb https://github.com/sponsors/codetheweb
- Applying database migrations...
✔ Database migrations applied.
- 📡 connecting to Discord...
✔ Ready! Invite the bot with https://discordapp.com/oauth2/authorize?client_id=226031561579954176&scope=bot%20applications.commands&permissions=36700160
<--- Last few GCs --->
[97:0x5d97bf0] 408753 ms: Mark-sweep 1886.6 (1946.4) -> 1874.6 (1940.0) MB, 716.3 / 0.0 ms (average mu = 0.251, current mu = 0.193) allocation failure; scavenge might not succeed
[97:0x5d97bf0] 409575 ms: Mark-sweep 1887.2 (1948.1) -> 1877.2 (1942.7) MB, 727.8 / 0.0 ms (average mu = 0.187, current mu = 0.115) allocation failure; scavenge might not succeed
<--- JS stacktrace --->
FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory
1: 0xb60cd0 node::Abort() [/usr/local/bin/node]
2: 0xa78155 node::FatalError(char const*, char const*) [/usr/local/bin/node]
3: 0xd392de v8::Utils::ReportOOMFailure(v8::internal::Isolate*, char const*, bool) [/usr/local/bin/node]
4: 0xd39657 v8::internal::V8::FatalProcessOutOfMemory(v8::internal::Isolate*, char const*, bool) [/usr/local/bin/node]
5: 0xf16715 [/usr/local/bin/node]
6: 0xf17618 v8::internal::Heap::RecomputeLimits(v8::internal::GarbageCollector) [/usr/local/bin/node]
7: 0xf27b23 [/usr/local/bin/node]
8: 0xf28998 v8::internal::Heap::CollectGarbage(v8::internal::AllocationSpace, v8::internal::GarbageCollectionReason, v8::GCCallbackFlags) [/usr/local/bin/node]
9: 0xf032fe v8::internal::HeapAllocator::AllocateRawWithLightRetrySlowPath(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment) [/usr/local/bin/node]
10: 0xf046c7 v8::internal::HeapAllocator::AllocateRawWithRetryOrFailSlowPath(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment) [/usr/local/bin/node]
11: 0xee4c10 v8::internal::Factory::AllocateRaw(int, v8::internal::AllocationType, v8::internal::AllocationAlignment) [/usr/local/bin/node]
12: 0xedc684 v8::internal::FactoryBase<v8::internal::Factory>::AllocateRawWithImmortalMap(int, v8::internal::AllocationType, v8::internal::Map, v8::internal::AllocationAlignment) [/usr/local/bin/node]
13: 0xedea18 v8::internal::FactoryBase<v8::internal::Factory>::NewRawTwoByteString(int, v8::internal::AllocationType) [/usr/local/bin/node]
14: 0x101ec32 v8::internal::JsonParser<unsigned short>::MakeString(v8::internal::JsonString const&, v8::internal::Handle<v8::internal::String>) [/usr/local/bin/node]
15: 0x10209a6 v8::internal::JsonParser<unsigned short>::ParseJsonValue() [/usr/local/bin/node]
16: 0x102148f v8::internal::JsonParser<unsigned short>::ParseJson() [/usr/local/bin/node]
17: 0xdbeb53 v8::internal::Builtin_JsonParse(int, unsigned long*, v8::internal::Isolate*) [/usr/local/bin/node]
18: 0x16dabb9 [/usr/local/bin/node]
npm notice
npm notice New major version of npm available! 8.15.0 -> 9.4.2
npm notice Changelog: <https://github.com/npm/cli/releases/tag/v9.4.2>
npm notice Run `npm install -g npm@9.4.2` to update!
npm notice
Done in 412.33s.
```

**Runtime**
I'm running Muse:
- [ ] Directly from the cloned repository
- [ x] Inside a Kubernetes Cluster (single pod) (Docker container)
- [ ] Something else (please elaborate)

**Versions**
- Muse: Latest
- Docker (if applicable):
- OS: Ubuntu 20.04 (RKE2)

**Comment by codetheweb** [comment]
Muse loads the entire playlist into memory, but it must be quite large to be crashing the bot. 😄 How many tracks are on the playlist?

I would recommend setting the environment variable `NODE_OPTIONS=--max-old-space-size=4096` (or any other larger setting than the default of 1.7gb I think) and see if that helps.

**Comment by HiImVINK** [comment]
uhmm 154 songs haha

ill try to set this environment variable. im sure that will work

**Comment by codetheweb** [comment]
I was able to reproduce this and it's now fixed in 2.1.9.
